e2e-tests/development-runtime/cypress/integration/hot-reloading/non-js-file.js

```diff
@@ -3,31 +3,63 @@ const TEST_ID = `sub-title`
 const message = `This is a sub-title`
 
 describe(`hot reloading non-js file`, () => {
-  beforeEach(() => {
-    cy.exec(
-      `npm run update -- --file content/2018-12-14-hello-world.md --replacements "${message}:%${TEMPLATE}%" --exact`
-    )
-    cy.wait(1000)
-
-    cy.visit(`/2018-12-14-hello-world/`).waitForRouteChange()
-
-    cy.wait(1000)
+  describe(`markdown`, () => {
+    beforeEach(() => {
+      cy.exec(
+        `npm run update -- --file content/2018-12-14-hello-world.md --replacements "${message}:%${TEMPLATE}%" --exact`
+      )
+      cy.wait(1000)
+  
+      cy.visit(`/2018-12-14-hello-world/`).waitForRouteChange()
+  
+      cy.wait(1000)
+    })
+  
+    it(`displays placeholder content on launch`, () => {
+      cy.getTestElement(TEST_ID).invoke(`text`).should(`contain`, TEMPLATE)
+    })
+  
+    it(`hot reloads with new content`, () => {
+      cy.getTestElement(TEST_ID).invoke(`text`).should(`contain`, TEMPLATE)
+  
+      cy.exec(
+        `npm run update -- --file content/2018-12-14-hello-world.md --replacements "${TEMPLATE}:${message}"`
+      )
+  
+      // wait for socket.io to update
+      cy.wait(5000)
+  
+      cy.getTestElement(TEST_ID).invoke(`text`).should(`eq`, message)
+    })
   })
 
-  it(`displays placeholder content on launch`, () => {
-    cy.getTestElement(TEST_ID).invoke(`text`).should(`contain`, TEMPLATE)
-  })
+  describe(`image`, () => {
+    beforeEach(() => {
+      cy.exec(
+        `npm run update -- --file src/images/image.png --copy "src/images/original.png"`
+      )
+      cy.wait(1000)
+  
+      cy.visit(`/md-image/`).waitForRouteChange()
+      cy.wait(1000)
+    })
 
-  it(`hot reloads with new content`, () => {
-    cy.getTestElement(TEST_ID).invoke(`text`).should(`contain`, TEMPLATE)
+    it(`displays original content on launch`, () => {
+      cy.matchImageSnapshot(`non-js-file--image-original`)
+    })
 
-    cy.exec(
-      `npm run update -- --file content/2018-12-14-hello-world.md --replacements "${TEMPLATE}:${message}"`
-    )
+    // This test currently has a snapshot of the "original" image, since gatsby-remark-images hasn't
+    // been updated in development-runtime to include the hot reloading fix for images.
+    // TODO: update gatsby-remark-images and update the snapshot for this test
```

Comment:
We need to address this TODO before the merge. We should run `gatsby-dev` on `e2e-tests/development-runtime` so the comitted snapshots is adjusted.

It's still good that we see what snapshots look before applying source code changes (AKA it shows image/page not being regenerated, confirming that this is good test for it)

Reply:
Good call. Looking at the `package.json` file in here, I didn't realize our e2e tests ran off of `gatsby-dev`. 

Updated!